### ISSUE TITLE
Can't use color name literals in struct property definitions

### ISSUE THREAD

**Issue by jturcotte** [sixtyfpsui/sixtyfps#897]
```javascript
struct S := {
    c: color,
}
Demo := Window {
    // OK
    property<color> c: red;
    // OK
    property<S> s1: {c: Colors.rgb(255,0,0)};
    // error: Unknown unqualified identifier 'red'
    property<S> s2: {c: red};
}
```

https://sixtyfps.io/releases/0.1.6/editor/?snippet=struct+S+%3A%3D+%7B%0A++++c%3A+color%2C%0A%7D%0ADemo+%3A%3D+Window+%7B%0A++++%2F%2F+OK%0A++++property%3Ccolor%3E+c%3A+red%3B%0A++++%2F%2F+OK%0A++++property%3CS%3E+s1%3A+%7Bc%3A+Colors.rgb%28255%2C0%2C0%29%7D%3B%0A++++%2F%2F+error%3A+Unknown+unqualified+identifier+%27red%27%0A++++property%3CS%3E+s2%3A+%7Bc%3A+red%7D%3B%0A%7D%0A

**Comment by ogoffart** [comment]
Yes, right now we only do "type-specific" lookup (colors, enums) depending of the type of the property itself.
Ideally we should use type inference and see the actual type of the expression we try to analyse.
We need to fix lookup rules anyway: #273

In the mean time, you can use `Colors.red` everywhere.

**Comment by jturcotte** [comment]
Would it make sense to write `Colors.xyz` in the documentation instead to prevent this surprise? I actually didn't know that color literals were defined in the Colors namespace.

I'm stumbled on [this](https://sixtyfps.io/releases/0.1.6/docs/rust/sixtyfps/docs/langref/index.html#colors-namespace) but I just noticed that there was also this section in the doc: https://sixtyfps.io/releases/0.1.6/docs/rust/sixtyfps/docs/langref/index.html#colors

**Comment by ogoffart** [comment]
Probably we should suggest to use `Colors.xyz`  or `EnumName.xyz` if xyz is not found and it is the value of some enum.

**Comment by ogoffart** [comment]
We should also support that for the `==`. (See issue #10867)
The problem with binary operator like that is that we expect some kind of symmetry, but with a naive approach, it would only work for the right hand side.
(We could do the lookup in two phases but that'd be complicated and potentially slow)
